src/main/java/game/Game.java

```diff
@@ -55,10 +59,21 @@ public void updateBoard() {
     }
 
     /**
-     * Updates the Player's PlayerState using the board's feedback based on the Player's current location. Also
-     * decreases the Player's temporary invincibility-frame-count by 1.
+     * Updates the Player's PlayerState using the board's feedback based on the Player's current location.
+     * Also scans through the list of boardObjects to add any ChasingElement to Player as an Observer.
+     * Also decreases the Player's temporary invincibility-frame-count by 1.
      */
     public void updatePlayerState() {
+
+        ArrayList<Element> boardObjects = this.getBoard().getObjectManager().getBoardObjects();
```

Comment:
Violates clean architecture and single responsibility.

-  Don't bring boards and object managers/objects (entities) into game (controller)
-  Game is responsible for connecting to player and board. Not for updating the playerstate.

This code actually belongs in objectManager, all this function should call is line 79 below.

Reply:
I agree this is problematic. My problem was that the Player needs to add every ChasingElement as its Observer, but ObjectManager is agnostic about Player. The only place where Player and board objects interact is in the Game....What do you suggest?

Reply:
Hmm I see, this is tough. I think the observer pattern itself breaks clean architecture since it needs to interact with  game. Let’s ask the TA.